site/guides/effective-go-krm-function.md

```diff
@@ -0,0 +1,212 @@
+# Effective GO KRM functions
+
+This guide gives tips to effectively write a KRM function. 
+
+This guide is for advanced kpt function users who find the [catalog.kpt.dev] cannot fulfil their needs 
+and want to design their own KRM functions. 
```

Comment:
I think `advanced kpt users` can confuse users and might discourage users to read further.

Maybe we can combine this line and previous line with something along the lines of ...

`
This guide describes best practices and patterns on how to structure your Go KRM functions.
`